pkg/sync/rpc/server.go

```diff
@@ -344,8 +345,8 @@ func (s *SyncAgentServer) FilesSync(ctx context.Context, req *ptypes.FilesSyncRe
 			ops = s.RebuildStatus
 		}
 
-		port, err := s.launchReceiver("FilesSync", info.ToFileName, ops)
-		if err != nil {
+		var port int
```

Comment:
It's very easy to make mistakes like this if it's done in this way.

Use the pattern in https://github.com/longhorn/longhorn-manager/blob/d2107aa19465040add77bd2198dd6b9a7cf4a109/api/volume.go#L20 instead.

Define the return valuable name in the function signature, so it can be used in the `defer func()` automatically.

Check other functions and mark the change if necessary.

Reply:
For `SnapshotPurge()`, it's better to define the variable `err` inside the goroutine. We don't need to modify this function.

Reply:
Yeah, that doesn't apply because it's a goroutine.

Reply:
In fact for `SnapshotPurge`, it's better to put the logic in the goroutine into a separate function. It's too long and can easily result in mistakes.